File: bench/py/fib_untyped_32.py
# Untyped recursive Fibonacci at n=32 - Python port of
# bench/qn/fib_untyped_32.qn (fib_typed's workload, untyped code: the row
# that makes Quoin's typed-vs-untyped gap directly comparable).
# Run: python3.13 bench/py/fib_untyped_32.py
#
# Kept as a class with a method (not a bare function) to preserve the
# dispatch-per-call shape of the Quoin original's `.value:` send.


class Fib:
    def value(self, n):
        if n <= 1:
            return n
        return self.value(n - 1) + self.value(n - 2)


r = Fib().value(32)
if r == 2178309:
    print('fib_untyped_32: ok')
else:
    print('fib_untyped_32: FAIL got ' + str(r))

File: bench/rb/fib_untyped_32.rb
# Untyped recursive Fibonacci at n=32 (fib_typed's workload) - in Quoin this
# row differs from fib_typed only in annotations, making the typed-vs-untyped
# gap directly readable. In Ruby every call is dynamic anyway, so this is the
# same code as fib_typed.rb; matching bench/qn/fib_untyped_32.qn.
# Run: `ruby bench/rb/fib_untyped_32.rb`.

class Fib
  def self.value(n)
    if n <= 1
      n
    else
      value(n - 1) + value(n - 2)
    end
  end
end

r = Fib.value(32)
if r == 2178309
  puts 'fib_untyped_32: ok'
else
  puts "fib_untyped_32: FAIL got #{r}"
end
